pkg/ccl/streamingccl/streamingest/ingest_span_configs.go

```diff
@@ -0,0 +1,249 @@
+// write updates and deletes with the same source side transaction commit
+// timestamp at the same new timestamp on the destination side. This invariant
+// ensures a span configuration's target (i.e. the span that a configuration
+// applies to) never overlaps with any other span configuration target. Else,
+// C2C would break the span config reconciliation system.
+//
+// TODO(msbutler): on an initial scan, we need to buffer up all updates and
+// write them to the span config table in one transaction, along with a delete
+// over the whole tenant key span. Since C2C does not lay a PTS on the source
+// side span config table, the initial scan on resumption may miss updates;
+// therefore, the only way to cleanly update the destination side span config
+// table is to write the latest state of the source table and delete all
+// existing state, in one transaction.
+type spanConfigIngestor struct {
+	accessor                    spanconfig.KVAccessor
+	bufferedUpdates             []spanconfig.Record
+	bufferedDeletes             []spanconfig.Target
+	lastBufferedSourceTimestamp hlc.Timestamp
+	session                     sqlliveness.Session
+	stopperCh                   chan struct{}
+	settings                    *cluster.Settings
+	client                      streamclient.Client
+	rekeyer                     *backupccl.KeyRewriter
+	testingKnobs                *sql.StreamingTestingKnobs
+}
+
+func makeSpanConfigIngestor(
+	ctx context.Context,
+	execCfg *sql.ExecutorConfig,
+	ingestionJob *jobs.Job,
+	sourceTenantID roachpb.TenantID,
+	stopperCh chan struct{},
+) (*spanConfigIngestor, error) {
+
+	client, err := connectToActiveClient(ctx, ingestionJob, execCfg.InternalDB, streamclient.ForSpanConfigs())
+	if err != nil {
+		return nil, err
+	}
+
+	details := ingestionJob.Details().(jobspb.StreamIngestionDetails)
+
+	rekeyCfg := execinfrapb.TenantRekey{
+		OldID: sourceTenantID,
+		NewID: details.DestinationTenantID,
+	}
+	rekeyer, err := backupccl.MakeKeyRewriterFromRekeys(keys.SystemSQLCodec,
+		nil /* tableRekeys */, []execinfrapb.TenantRekey{rekeyCfg},
+		true /* restoreTenantFromStream */)
+	if err != nil {
+		return nil, err
+	}
+	log.Infof(ctx, "initialized span config ingestor")
+	return &spanConfigIngestor{
+		accessor:     execCfg.SpanConfigKVAccessor,
+		settings:     execCfg.Settings,
+		session:      ingestionJob.Session(),
+		client:       client,
+		rekeyer:      rekeyer,
+		stopperCh:    stopperCh,
+		testingKnobs: execCfg.StreamingTestingKnobs,
+	}, nil
+}
+
+func (sc *spanConfigIngestor) ingestSpanConfigs(
+	ctx context.Context, tenantName roachpb.TenantName,
+) error {
+	sub, err := sc.client.SetupSpanConfigsStream(ctx, tenantName)
+	if err != nil {
+		return err
+	}
+
+	group := ctxgroup.WithContext(ctx)
+	group.GoCtx(sub.Subscribe)
+	group.GoCtx(func(ctx context.Context) error {
+		defer func() {
+			if err := sc.client.Close(ctx); err != nil {
+				log.Errorf(ctx, "error on client.close(): %s", err)
+			}
+		}()
```

Comment:
If you fancy, I think there is a closeAndLog(ctx, client) in this package

Reply:
neat!